docs/Unity-Environment-Registry.md

```diff
@@ -0,0 +1,60 @@
+# Unity Environment Registry [Experimental]
+
+The Unity Environment Registry is a database of pre-built Unity environments that can be easily used without having to install the Unity Editor. It is a great way to get started with our [UnityEnvironment API](Python-API.md).
+
+## Loading an Environment from the Registry
+
+To get started, you can access the default registry we provide with our [Example Environments](Learning-Environment-Examples.md). The Unity Environment Registry implements a _Mapping_, therefore, you can access an entry with its identifier with the square brackets `[ ]`. Use the following code to list all of the environment identifiers present in the default registry:
+
+```python
+from mlagents_envs.registry import default_registry
+
+environment_names = list(default_registry.keys())
+for name in environment_names:
+   print(name)
+```
+
+The `make()` method on a registry value will return a `UnityEnvironment` ready to be used. All arguments passed to the make method will be passed to the constructor of the `UnityEnvironment` as well. Refer to the documentation on the [Python-API](Python-API.md) for more information about the arguments of the `UnityEnvironment` constructor. For example, the following code will create the environment under the identifier `"my-env"`, reset it, perform a few steps and finally close it:
+
+```python
+from mlagents_envs.registry import default_registry
+
+env = default_registry["my-env"].make()
+env.reset()
+for _ in range(10):
+  env.step()
+env.close()
+```
+
+## Create and share your own registry
+
+In order to share the `UnityEnvironemnt` you created, you must :
+ - Create a Unity executable of your environment for each platform (Linux, OSX and/or Windows)
+ - Place each executable in a `zip` compressed folder
+ - Upload each zip file online
+ - Create a `yaml` file that will contain the description and path to your environment
+ - Upload the `yaml` file online
+The `yaml` file must have the following format :
+
+```yaml
+environments:
+  - <environment-identifier>:
+     expected_reward: <expected-reward-float>
+     description: <description-of-the-environment>
+     linux_url: <url-to-the-linux-zip-folder>
+     darwin_url: <url-to-the-osx-zip-folder>
+     win_url: <url-to-the-windows-zip-folder>
+     additional_args:
+      - <an-optional-list-of-command-line-arguments-for-the-executable>
+      - ...
+```
+
+Your users can now use your environment with the following code :
```

Comment:
Should it be made clear that they are not pushing these builds to our storage but they must host it on their own? Is there another step to use the new env i.e. `pip install foo-env` in gym or is this beyond the scope of this WIP?

Reply:
There are no additional step. No need to pip install anything, just get the url of the yaml

Reply:
So if you have a new env do you submit a PR to our repo to have it added to the registry? Or do you need to go to a fork?

Reply:
Ok, so it seems the documentation is not clear. Please advise on how to make more explicit.
If you made a brand new environment, you need to : 
 - Compile it into an executable
 - Zip the executable
 - Upload the executable to YOUR OWN hosting service (example : gcs)
 - Create a yaml file that contains the url to the zip file
 - Upload the yaml file to hosting service

Once this is done, to share your environment, all you need to do is advertise the URL to the yaml. If a user wants your environment, all they have to do is call

```
env_registry = UnityEnvRegistry()
env_registry.register_from_yaml("your-url-here")
```
And they will have your environment.
No PR, no fork, no pip

Reply:
@andrew is this confusing? Should I change the documentation to give more details ?

Reply:
I think that makes sense. I was being dense.  Also, thats not me^

Reply:
😨